File: Sort/MergeSort/MergeSort.c
#include <stdio.h>

void sort(int *arreglo, int tam);

void mezcla(int *arrIzq, int tamIzq, int *arrDer, int tamDer, int* arreglo);


void sort(int *arreglo, int tam){
    if(tam>1){
        int i,j,k;
        int mitad=tam/2;
        int arrIzq[mitad];
        int arrDer[tam-mitad];
        for(i=0, k=0; i<mitad; i++, k++){
            arrIzq[i]=arreglo[k];
        }
        for(j=0; j< (tam-mitad); j++, k++){
            arrDer[j]=arreglo[k];
        }
        sort(arrIzq,mitad);
        sort(arrDer,tam-mitad);
        mezcla(arrIzq,mitad, arrDer,tam-mitad,arreglo);
    }
}

void mezcla(int *arrIzq, int tamIzq, int *arrDer, int tamDer, int* arreglo){
    int i=0,j=0,k=0;
    while(i<tamIzq && j<tamDer){
        if(arrIzq[i]<arrDer[j]){
            arreglo[k]=arrIzq[i];
            i++;   
        } else{
            arreglo[k]=arrDer[j];
            j++;  
        }
        k++;
    }
    while(j<tamDer){
        arreglo[k]=arrDer[j];
        j++;
        k++;
    }
    while(i<tamIzq){
        arreglo[k]=arrIzq[i];
        i++;
        k++;
    }
}


void main(){
    puts("Inicio");
    int arreglo[4]={3,2,1,0};
    sort(arreglo,4);
    for (int i=0;i<4;i++){
        printf("%d ",arreglo[i]);
    }
    puts("\nFin");
}

File: Sort/MergeSort/MergeSort.py
def run():
    pass

def Sort(arr:list, tam:int):
    if(tam>1):
        i , j , k = 0
        mitad = tam/2
        arrIzq[mitad] = 
        #Si sobran elementos en el arreglo de la izquierda los almacena en el arreglo original
        for i in range(0,mitad):
            arrIzq[i] = arreglo[k]
            i+=1
            k+=1

        #Si sobran elementos en el arreglo de la derecha los almacena en el arreglo original
        for i in range(0,tam-mitad):
            arrDer[j] = arreglo[k]
            j+=1
            k+=1
        Sort(arrIzq,mitad)
        Sort(arrDer,tam-mitad)
        Mezcla(arrIzq,mitad,arrDer,tam-mitad,arr)

def Mezcla(arrIzq,tamIzq,arrDer,tamDer,arr):
    pass

if __name__ == '__main__':
    run()